planner/core/plan_cost_detail.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import "fmt"
+
+const (
+	// RowSizeLbl indicates for rowSize
+	RowSizeLbl = "rowSize"
+	// NetworkFactorLbl indicates for networkFactor
+	NetworkFactorLbl = "networkFactor"
+	// SeekFactorLbl indicates for seekFactor
+	SeekFactorLbl = "seekFactor"
+)
+
+func setPointGetPlanCostDetail(p *PointGetPlan, opt *physicalOptimizeOp,
+	rowSize, networkFactor, seekFactor float64) {
+	if opt == nil {
+		return
+	}
+	opt.appendPlanCostDetail(p).
+		appendParamCostDetail(RowSizeLbl, rowSize).
+		appendParamCostDetail(NetworkFactorLbl, networkFactor).
+		appendParamCostDetail(SeekFactorLbl, seekFactor).
+		appendCurrCostDesc(fmt.Sprintf("%s * %s + %s", RowSizeLbl, NetworkFactorLbl, SeekFactorLbl))
```

Comment:
I don't get why we need this logic.
I think just inserting the new `PhysicalPlanCostDetail` into the `map[int]*PhysicalPlanCostDetail` has the same effect and is much simpler.

Reply:
updated